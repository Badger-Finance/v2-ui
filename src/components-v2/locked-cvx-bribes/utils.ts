import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
