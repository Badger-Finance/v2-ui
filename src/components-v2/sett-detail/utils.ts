import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
