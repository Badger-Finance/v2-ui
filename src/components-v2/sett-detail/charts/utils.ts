import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
