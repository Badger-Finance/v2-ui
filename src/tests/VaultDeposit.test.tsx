import '@testing-library/jest-dom';

import userEvent from '@testing-library/user-event';
import { StoreProvider } from 'mobx/stores/store-context';
import React from 'react';

import { VaultDeposit } from '../components-v2/common/dialogs/VaultDeposit';
import store from '../mobx/stores/RootStore';
import { customRender, screen } from './Utils';
import { SAMPLE_VAULT } from './utils/samples';

describe('Vault Deposit', () => {
  test('displays sett information', () => {
    const { baseElement } = customRender(
      <StoreProvider value={store}>
        <VaultDeposit open={true} vault={SAMPLE_VAULT} onClose={jest.fn()} />
      </StoreProvider>,
    );
    expect(baseElement).toMatchSnapshot();
  });

  test('can see full fees descriptions', () => {
    const vault = { ...SAMPLE_VAULT };
    vault.strategy.strategistFee = 0.3;
    vault.strategy.address = '0x9BE89D2a4cd102D8Fecc6BF9dA793be995C22541';
    const { baseElement } = customRender(
      <StoreProvider value={store}>
        <VaultDeposit open={true} vault={vault} onClose={jest.fn()} />
      </StoreProvider>,
    );
    userEvent.click(screen.getAllByTitle(/click to see full description/i)[0]);
    expect(baseElement).toMatchSnapshot();
  });

  test('can go back from full fees descriptions', () => {
    const vault = { ...SAMPLE_VAULT };
    vault.strategy.strategistFee = 0.3;
    vault.strategy.address = '0x9BE89D2a4cd102D8Fecc6BF9dA793be995C22541';
    const { baseElement } = customRender(
      <StoreProvider value={store}>
        <VaultDeposit open={true} vault={vault} onClose={jest.fn()} />
      </StoreProvider>,
    );
    userEvent.click(screen.getAllByTitle(/click to see full description/i)[0]);
    userEvent.click(screen.getByRole('button', { name: 'Back' }));
    expect(baseElement).toMatchSnapshot();
  });
});
