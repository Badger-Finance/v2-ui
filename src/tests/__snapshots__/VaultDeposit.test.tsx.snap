// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vault Deposit can go back from full fees descriptions 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root-13"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root-28"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container-16 MuiDialog-scrollPaper-14"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root-30 MuiDialog-paper-17 MuiDialog-paperScrollPaper-18 MuiDialog-paperWidthXl-25 MuiDialog-paperFullWidth-26 MuiPaper-elevation24-57 MuiPaper-rounded-31"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root-375 makeStyles-root-371"
        >
          <h2
            class="MuiTypography-root-63 MuiTypography-h6-73"
          >
            <div
              class="MuiGrid-root-93 MuiGrid-container-94 MuiGrid-align-items-xs-center-102"
            >
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
              >
                <p
                  class="MuiTypography-root-63 makeStyles-mode-372 MuiTypography-body1-65 MuiTypography-colorTextSecondary-89"
                >
                  Deposit
                </p>
              </div>
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
              >
                <div
                  class="MuiBox-root-376 MuiBox-root-377"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-198 makeStyles-closeIcon-374"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </h2>
        </div>
        <div
          class="MuiDialogContent-root-378 makeStyles-content-1"
        >
          <div
            class="MuiBox-root-376 MuiBox-root-380 makeStyles-tokenBox-12"
          >
            <p
              class="MuiTypography-root-63 token-label MuiTypography-body1-65"
            >
              Token
            </p>
            <div
              class="MuiGrid-root-93 MuiGrid-container-94"
            >
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-7-148"
              >
                <div
                  class="MuiBox-root-376 MuiBox-root-381 token-logo-name"
                >
                  <div
                    class="makeStyles-root-382"
                  >
                    <img
                      alt="BBTC logo"
                      class="makeStyles-position-383"
                      src="/assets/icons/bbtc.svg"
                    />
                    <img
                      alt="crvRenWSBTC logo"
                      class="makeStyles-position-384"
                      src="/assets/icons/crvrenwsbtc.svg"
                    />
                  </div>
                  <p
                    class="MuiTypography-root-63 token-name MuiTypography-body1-65"
                  >
                    bBTC/sbtcCRV
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-5-146"
              >
                <div
                  class="MuiFormControl-root-386 MuiTextField-root-385 makeStyles-amountTextField-6  MuiFormControl-fullWidth-389"
                >
                  <div
                    class="MuiInputBase-root-405 MuiOutlinedInput-root-390 MuiInputBase-fullWidth-415 MuiInputBase-formControl-406"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input-416 MuiOutlinedInput-input-400"
                      inputmode="decimal"
                      maxlength="79"
                      minlength="1"
                      pattern="^[0-9]*[.,]?[0-9]*$"
                      placeholder="Type an amount to deposit"
                      spellcheck="false"
                      style="text-align: right;"
                      type="text"
                      value="0"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-423 MuiOutlinedInput-notchedOutline-399"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-424"
                        style="width: 0.01px;"
                      >
                        <span>
                          â€‹
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiBox-root-376 MuiBox-root-427 makeStyles-amountDollarValue-8"
                >
                  <p
                    class="MuiTypography-root-63 MuiTypography-body1-65"
                  >
                    ~$
                    0.00
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root-93 token-balance-percentage MuiGrid-container-94 MuiGrid-align-items-xs-center-102"
            >
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-6-147"
              >
                <p
                  class="MuiTypography-root-63 token-balance MuiTypography-body1-65 MuiTypography-colorTextSecondary-89"
                >
                  Balance: 0.0
                </p>
              </div>
              <div
                class="MuiGrid-root-93 WithStyles(ForwardRef(Grid))-root-428 WithStyles(ForwardRef(Grid))-root-429 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-6-147"
              >
                <div
                  class="MuiButtonGroup-root-431"
                  role="group"
                >
                  <button
                    aria-label="25%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-437 MuiButtonGroup-groupedHorizontal-438 MuiButtonGroup-groupedOutlined-445 MuiButtonGroup-groupedOutlinedHorizontal-446 MuiButtonGroup-groupedOutlined-445 makeStyles-button-430 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      25
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-479"
                    />
                  </button>
                  <button
                    aria-label="50%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-437 MuiButtonGroup-groupedHorizontal-438 MuiButtonGroup-groupedOutlined-445 MuiButtonGroup-groupedOutlinedHorizontal-446 MuiButtonGroup-groupedOutlined-445 makeStyles-button-430 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      50
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-479"
                    />
                  </button>
                  <button
                    aria-label="75%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-437 MuiButtonGroup-groupedHorizontal-438 MuiButtonGroup-groupedOutlined-445 MuiButtonGroup-groupedOutlinedHorizontal-446 MuiButtonGroup-groupedOutlined-445 makeStyles-button-430 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      75
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-479"
                    />
                  </button>
                  <button
                    aria-label="100%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-437 MuiButtonGroup-groupedHorizontal-438 MuiButtonGroup-groupedOutlined-445 MuiButtonGroup-groupedOutlinedHorizontal-446 MuiButtonGroup-groupedOutlined-445 makeStyles-button-430 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      100
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-479"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-fees-3"
          >
            <div
              class="MuiGrid-root-93 MuiGrid-container-94"
            >
              <div
                class="MuiGrid-root-93 makeStyles-feeRow-459 MuiGrid-container-94 MuiGrid-justify-content-xs-space-between-113"
              >
                <div
                  class="MuiBox-root-376 MuiBox-root-460"
                >
                  <span
                    class="MuiTypography-root-63 makeStyles-specName-457 MuiTypography-body1-65 MuiTypography-colorTextSecondary-89"
                  >
                    Performance Fee
                     
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="see fees descriptions"
                    class="MuiSvgIcon-root-198 makeStyles-helpIcon-458 MuiSvgIcon-colorPrimary-199"
                    focusable="false"
                    title="Click to see full description"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                    />
                  </svg>
                </div>
                <h6
                  class="MuiTypography-root-63 MuiTypography-subtitle2-75 MuiTypography-displayInline-91"
                >
                  20%
                </h6>
              </div>
              <div
                class="MuiGrid-root-93 makeStyles-feeRow-459 MuiGrid-container-94 MuiGrid-justify-content-xs-space-between-113"
              >
                <div
                  class="MuiBox-root-376 MuiBox-root-472"
                >
                  <span
                    class="MuiTypography-root-63 makeStyles-specName-457 MuiTypography-body1-65 MuiTypography-colorTextSecondary-89"
                  >
                    Withdraw Fee
                     
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="see fees descriptions"
                    class="MuiSvgIcon-root-198 makeStyles-helpIcon-458 MuiSvgIcon-colorPrimary-199"
                    focusable="false"
                    title="Click to see full description"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                    />
                  </svg>
                </div>
                <h6
                  class="MuiTypography-root-63 MuiTypography-subtitle2-75 MuiTypography-displayInline-91"
                >
                  0.1%
                </h6>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-93 makeStyles-totalAmountContainer-9 MuiGrid-container-94"
          >
            <div
              class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
            >
              <div
                class="MuiBox-root-376 MuiBox-root-473 makeStyles-totalAmountLabel-10"
              >
                Total Deposit
              </div>
            </div>
            <div
              class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
            >
              <div
                class="MuiBox-root-376 MuiBox-root-474 makeStyles-totalAmount-11"
              >
                0
              </div>
            </div>
          </div>
          <button
            aria-label="Deposit"
            class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-contained-292 WithStyles(ForwardRef(Button))-root-477 WithStyles(ForwardRef(Button))-root-478 Styled(WithStyles(ForwardRef(Button)))-root-475 Styled(WithStyles(ForwardRef(Button)))-root-476 MuiButton-containedPrimary-293 MuiButton-containedSizeLarge-304 MuiButton-sizeLarge-306 MuiButton-disabled-297 MuiButton-fullWidth-307 MuiButtonBase-disabled-314"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label-285"
            >
              DEPOSIT bBTC/sbtcCRV
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Vault Deposit can see full fees descriptions 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root-13"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root-28"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container-16 MuiDialog-scrollPaper-14"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root-30 MuiDialog-paper-17 MuiDialog-paperScrollPaper-18 MuiDialog-paperWidthXl-25 MuiDialog-paperFullWidth-26 MuiPaper-elevation24-57 MuiPaper-rounded-31"
        role="dialog"
      >
        <div
          class="MuiGrid-root-93 makeStyles-root-350 MuiGrid-container-94"
        >
          <div
            class="MuiGrid-root-93 MuiGrid-container-94 MuiGrid-align-items-xs-center-102 MuiGrid-justify-content-xs-space-between-113"
          >
            <p
              class="MuiTypography-root-63 makeStyles-title-351 MuiTypography-body1-65 MuiTypography-displayInline-91"
            >
              Fee Information
            </p>
            <button
              class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-text-286 makeStyles-backButton-4 MuiButton-textPrimary-287"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-285"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-198 makeStyles-arrowIcon-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
                  />
                </svg>
                Back
              </span>
              <span
                class="MuiTouchRipple-root-361"
              />
            </button>
          </div>
          <div
            class="makeStyles-content-352"
          >
            <p
              class="MuiTypography-root-63 MuiTypography-body1-65 MuiTypography-colorTextSecondary-89"
            >
              Fees are charged on certain Sett Vaults, and are sent to the BadgerDAO treasury, or shared with the strategist who wrote the vault strategy.
            </p>
            <hr
              class="MuiDivider-root-354 makeStyles-divider-353"
            />
            <div>
              <p
                class="MuiTypography-root-63 MuiTypography-body2-64 MuiTypography-colorTextSecondary-89"
              >
                - Performance Fees are assessed on the profit that the strategy creates
              </p>
              <p
                class="MuiTypography-root-63 MuiTypography-body2-64 MuiTypography-colorTextSecondary-89"
              >
                - Management Fees are charged on the entire principal over the course of a year
              </p>
              <p
                class="MuiTypography-root-63 MuiTypography-body2-64 MuiTypography-colorTextSecondary-89"
              >
                - Both Performance and Management Fees are factored into all ROI calculations
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Vault Deposit displays sett information 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root-13"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root-28"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container-16 MuiDialog-scrollPaper-14"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root-30 MuiDialog-paper-17 MuiDialog-paperScrollPaper-18 MuiDialog-paperWidthXl-25 MuiDialog-paperFullWidth-26 MuiPaper-elevation24-57 MuiPaper-rounded-31"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root-62 makeStyles-root-58"
        >
          <h2
            class="MuiTypography-root-63 MuiTypography-h6-73"
          >
            <div
              class="MuiGrid-root-93 MuiGrid-container-94 MuiGrid-align-items-xs-center-102"
            >
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
              >
                <p
                  class="MuiTypography-root-63 makeStyles-mode-59 MuiTypography-body1-65 MuiTypography-colorTextSecondary-89"
                >
                  Deposit
                </p>
              </div>
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
              >
                <div
                  class="MuiBox-root-196 MuiBox-root-197"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-198 makeStyles-closeIcon-61"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </h2>
        </div>
        <div
          class="MuiDialogContent-root-207 makeStyles-content-1"
        >
          <div
            class="MuiBox-root-196 MuiBox-root-209 makeStyles-tokenBox-12"
          >
            <p
              class="MuiTypography-root-63 token-label MuiTypography-body1-65"
            >
              Token
            </p>
            <div
              class="MuiGrid-root-93 MuiGrid-container-94"
            >
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-7-148"
              >
                <div
                  class="MuiBox-root-196 MuiBox-root-210 token-logo-name"
                >
                  <div
                    class="makeStyles-root-211"
                  >
                    <img
                      alt="BBTC logo"
                      class="makeStyles-position-212"
                      src="/assets/icons/bbtc.svg"
                    />
                    <img
                      alt="crvRenWSBTC logo"
                      class="makeStyles-position-213"
                      src="/assets/icons/crvrenwsbtc.svg"
                    />
                  </div>
                  <p
                    class="MuiTypography-root-63 token-name MuiTypography-body1-65"
                  >
                    bBTC/sbtcCRV
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-5-146"
              >
                <div
                  class="MuiFormControl-root-215 MuiTextField-root-214 makeStyles-amountTextField-6  MuiFormControl-fullWidth-218"
                >
                  <div
                    class="MuiInputBase-root-234 MuiOutlinedInput-root-219 MuiInputBase-fullWidth-244 MuiInputBase-formControl-235"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input-245 MuiOutlinedInput-input-229"
                      inputmode="decimal"
                      maxlength="79"
                      minlength="1"
                      pattern="^[0-9]*[.,]?[0-9]*$"
                      placeholder="Type an amount to deposit"
                      spellcheck="false"
                      style="text-align: right;"
                      type="text"
                      value="0"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-252 MuiOutlinedInput-notchedOutline-228"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-253"
                        style="width: 0.01px;"
                      >
                        <span>
                          â€‹
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiBox-root-196 MuiBox-root-256 makeStyles-amountDollarValue-8"
                >
                  <p
                    class="MuiTypography-root-63 MuiTypography-body1-65"
                  >
                    ~$
                    0.00
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root-93 token-balance-percentage MuiGrid-container-94 MuiGrid-align-items-xs-center-102"
            >
              <div
                class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-6-147"
              >
                <p
                  class="MuiTypography-root-63 token-balance MuiTypography-body1-65 MuiTypography-colorTextSecondary-89"
                >
                  Balance: 0.0
                </p>
              </div>
              <div
                class="MuiGrid-root-93 WithStyles(ForwardRef(Grid))-root-257 WithStyles(ForwardRef(Grid))-root-258 MuiGrid-item-95 MuiGrid-grid-xs-12-139 MuiGrid-grid-sm-6-147"
              >
                <div
                  class="MuiButtonGroup-root-260"
                  role="group"
                >
                  <button
                    aria-label="25%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-266 MuiButtonGroup-groupedHorizontal-267 MuiButtonGroup-groupedOutlined-274 MuiButtonGroup-groupedOutlinedHorizontal-275 MuiButtonGroup-groupedOutlined-274 makeStyles-button-259 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      25
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-324"
                    />
                  </button>
                  <button
                    aria-label="50%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-266 MuiButtonGroup-groupedHorizontal-267 MuiButtonGroup-groupedOutlined-274 MuiButtonGroup-groupedOutlinedHorizontal-275 MuiButtonGroup-groupedOutlined-274 makeStyles-button-259 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      50
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-324"
                    />
                  </button>
                  <button
                    aria-label="75%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-266 MuiButtonGroup-groupedHorizontal-267 MuiButtonGroup-groupedOutlined-274 MuiButtonGroup-groupedOutlinedHorizontal-275 MuiButtonGroup-groupedOutlined-274 makeStyles-button-259 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      75
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-324"
                    />
                  </button>
                  <button
                    aria-label="100%"
                    class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-outlined-289 MuiButtonGroup-grouped-266 MuiButtonGroup-groupedHorizontal-267 MuiButtonGroup-groupedOutlined-274 MuiButtonGroup-groupedOutlinedHorizontal-275 MuiButtonGroup-groupedOutlined-274 makeStyles-button-259 MuiButton-outlinedSizeSmall-301 MuiButton-sizeSmall-305"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-285"
                    >
                      100
                      %
                    </span>
                    <span
                      class="MuiTouchRipple-root-324"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-fees-3"
          >
            <p
              class="MuiTypography-root-63 makeStyles-specName-316 MuiTypography-body1-65 MuiTypography-colorTextSecondary-89 MuiTypography-displayInline-91"
            >
              There are no fees for this vault
            </p>
          </div>
          <div
            class="MuiGrid-root-93 makeStyles-totalAmountContainer-9 MuiGrid-container-94"
          >
            <div
              class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
            >
              <div
                class="MuiBox-root-196 MuiBox-root-318 makeStyles-totalAmountLabel-10"
              >
                Total Deposit
              </div>
            </div>
            <div
              class="MuiGrid-root-93 MuiGrid-item-95 MuiGrid-grid-xs-6-133"
            >
              <div
                class="MuiBox-root-196 MuiBox-root-319 makeStyles-totalAmount-11"
              >
                0
              </div>
            </div>
          </div>
          <button
            aria-label="Deposit"
            class="MuiButtonBase-root-313 MuiButton-root-284 MuiButton-contained-292 WithStyles(ForwardRef(Button))-root-322 WithStyles(ForwardRef(Button))-root-323 Styled(WithStyles(ForwardRef(Button)))-root-320 Styled(WithStyles(ForwardRef(Button)))-root-321 MuiButton-containedPrimary-293 MuiButton-containedSizeLarge-304 MuiButton-sizeLarge-306 MuiButton-disabled-297 MuiButton-fullWidth-307 MuiButtonBase-disabled-314"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label-285"
            >
              DEPOSIT bBTC/sbtcCRV
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
